### DISCUSSION TITLE
Allow player to suppress certain types of Adviser messages

### DISCUSSION THREAD

**Discussion by MandrakeHorse** [CorsixTH/CorsixTH#2662] [Research Room]
### What feature or enhancement would you like to add?

In both Theme Hospital and CorsixTH, the Adviser can only be enabled or disabled, resulting in an all-or-nothing approach. One way to improve the Adviser would be to allow the player to configure what kinds of messages they want to receive from him. For example, if they don't want to receive updates on level progress, general praise, hiring advice, or guidance while building, they can suppress these message types so that they never see them. (Important messages such as Earthquake Alerts, VIP/Emergency arrivals could be exempt so that they always appear.)

This approach would involve a configuration window with a list of message types, each with a toggle that enables or disables them. All the message types would be enabled by default. This window would also contain the setting that turns the Adviser on and off.

### How would this feature benefit the gameplay of CorsixTH?

The player can select what types of messages they want to receive from the Adviser and can prevent message types that they don't want to see.
The player can configure the Adviser to their preference, rather than having him either enabled or disabled.

### Was the feature in the original game?

No

### What version of CorsixTH are you using?

0.68

### Does this feature/enhancement relate to a current problem?

No

**Comment by ARGAMX** [comment]
As an example how game notification can be tuned (not only Advisor messages, but also Fax):

<img width="479" alt="Screenshot_2024-10-07_at_12 22 13" src="https://github.com/user-attachments/assets/8143564b-ddef-420c-900d-f91c91caaefe">

But of course it would be better if user could be able to select between these options by itself.

This functionality connected with #2661.

**Reply by ARGAMX** [reply]
> <img alt="Screenshot_2024-10-07_at_12 22 13" width="479" src="https://private-user-images.githubusercontent.com/11425173/374484449-8143564b-ddef-420c-900d-f91c91caaefe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ1Njk4ODAsIm5iZiI6MTc1NDU2OTU4MCwicGF0aCI6Ii8xMTQyNTE3My8zNzQ0ODQ0NDktODE0MzU2NGItZGRlZi00MjBjLTkwMGQtZjkxYzkxY2FhZWZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA3VDEyMjYyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU5MGZmNWY0ZWJjNTViODIzYWUyODUzZTY0ZTI0ZTg2NDBjMGU0MWI4OTZmOTM3NGQ0MGNkODE4NzQxYjhhYzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4_5ESpM7KZFecWIFQjji6CVL40w8IAEeI0R9gCQFgKw">

Also `Research category finished` event belongs to this list.
Currently just Advisor informs about this event.
